pkg/server/api/entry_test.go

```diff
@@ -95,3 +95,120 @@ func TestRegistrationEntryToProto(t *testing.T) {
+				Ttl:      60,
+				Selectors: []*types.Selector{
+					{Type: "unix", Value: "uid:1000"},
+					{Type: "unix", Value: "gid:1000"},
+				},
+				FederatesWith: []string{
+					"spiffe://domain1.com",
+					"domain2.com",
+				},
+				Admin:      true,
+				ExpiresAt:  expiresAt,
+				DnsNames:   []string{"dns1", "dns2"},
+				Downstream: true,
+			},
+			expectEntry: &common.RegistrationEntry{
+				EntryId:  "entry1",
+				ParentId: "spiffe://example.org/foo",
+				SpiffeId: "spiffe://example.org/bar",
+				Ttl:      60,
+				Selectors: []*common.Selector{
+					{Type: "unix", Value: "uid:1000"},
+					{Type: "unix", Value: "gid:1000"},
+				},
+				FederatesWith: []string{
+					"spiffe://domain1.com",
+					"spiffe://domain2.com",
+				},
+				Admin:       true,
+				EntryExpiry: expiresAt,
+				DnsNames:    []string{"dns1", "dns2"},
+				Downstream:  true,
+			},
+		},
+		{
+			name: "missing entry",
+			err:  "missing entry",
+		},
+		{
+			name: "no parent ID",
+			err:  "invalid parent ID: request must specify SPIFFE ID",
+			entry: &types.Entry{
+				SpiffeId: &types.SPIFFEID{TrustDomain: "example.org", Path: "/bar"},
+			},
+		},
+		{
+			name: "malformed parent ID",
+			err:  "invalid parent ID: spiffeid: unable to parse:",
+			entry: &types.Entry{
+				ParentId: &types.SPIFFEID{TrustDomain: "invalid domain"},
+				SpiffeId: &types.SPIFFEID{TrustDomain: "example.org", Path: "/bar"},
+			},
+		},
+		{
+			name: "no spiffe ID",
+			err:  "invalid spiffe ID: request must specify SPIFFE ID",
+			entry: &types.Entry{
+				ParentId: &types.SPIFFEID{TrustDomain: "example.org", Path: "/bar"},
+			},
+		},
+		{
+			name: "malformed spiffe ID",
+			err:  "invalid spiffe ID: spiffeid: unable to parse:",
+			entry: &types.Entry{
+				SpiffeId: &types.SPIFFEID{TrustDomain: "invalid domain"},
+				ParentId: &types.SPIFFEID{TrustDomain: "example.org", Path: "/bar"},
+			},
+		},
+		{
+			name: "invalid DNS name",
+			err:  "invalid DNS name: label does not match regex: abc-",
+			entry: &types.Entry{
+				SpiffeId: &types.SPIFFEID{TrustDomain: "example.org", Path: "/foo"},
+				ParentId: &types.SPIFFEID{TrustDomain: "example.org", Path: "/bar"},
+				DnsNames: []string{"abc-"},
+			},
+		},
+		{
+			name: "malformed federated trust domain",
+			err:  "invalid federated trust domain: spiffeid: unable to parse: parse spiffe://malformed td:",
+			entry: &types.Entry{
+				SpiffeId:      &types.SPIFFEID{TrustDomain: "example.org", Path: "/foo"},
+				ParentId:      &types.SPIFFEID{TrustDomain: "example.org", Path: "/bar"},
+				FederatesWith: []string{"malformed td"},
+			},
+		},
+	} {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			entry, err := api.ProtoToRegistrationEntry(tt.entry)
+			if tt.err != "" {
+				require.Contains(t, err.Error(), tt.err)
+				require.Nil(t, entry)
+
+				return
+			}
+
+			require.NoError(t, err)
+			spiretest.AssertProtoEqual(t, tt.expectEntry, entry)
+		})
+	}
+}
```

Comment:
Do we want to have a test for createEntry? (I know that's harder to do)

Reply:
I don't think so. In general only the exported interface should be unit tested unless circumstances make it hard to test all of the internals through the exported interface. Considering the BatchCreateEntry is just a looping shim around `createEntry` and its unit tests pretty comprehensively unit test the internal behavior, I think we're ok?